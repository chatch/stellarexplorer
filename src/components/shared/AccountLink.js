import React from 'react'
import PropTypes from 'prop-types'
import {Link} from 'react-router-dom'
import has from 'lodash/has'

import knownAccounts from '../../data/known_accounts'

const AccountLink = ({account, label, hideKnown = false}) => {
  if (!account || account == null) return null
  let accLabel = label
  if (!accLabel) {
    accLabel =
      has(knownAccounts, account) && !hideKnown ? (
        <span style={{fontStyle: 'italic'}}>{knownAccounts[account].name}</span>
      ) : (
        account.substring(0, 4)
      )
  }
  return (
    <span title={account}>
      <Link to={`/account/${account}`}>{accLabel}</Link>
    </span>
  )
}

AccountLink.propTypes = {
  account: PropTypes.string.isRequired,
  hideKnown: PropTypes.bool,
  label: PropTypes.string,
}

export default AccountLink
