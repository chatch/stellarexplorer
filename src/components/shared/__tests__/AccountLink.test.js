import React from 'react'
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import knownAccounts from '../../../data/known_accounts'
import AccountLink from '../AccountLink'

configure({adapter: new Adapter()})


const ACC_KNOWN = Object.keys(knownAccounts).find(key => knownAccounts[key].displayName === 'NaoBTC')
const ACC_UNKNOWN = 'GCGG3CIRBG2TTBR4HYZJ7JLDRFKZIYOAHFXRWLU62CA2QN52P2SUQNPJ'
const LABEL = 'Anchor'

it('renders with label', () => {
  const link = shallow(<AccountLink label={LABEL} account={ACC_UNKNOWN} />)
  expect(link.getElements()).toMatchSnapshot()
})

it('renders short account for label when no label property', () => {
  const link = shallow(<AccountLink account={ACC_UNKNOWN} />)
  expect(link.getElements()).toMatchSnapshot()
})

it('renders anchor name in italics if account a known anchor', () => {
  const anchorName = knownAccounts[ACC_KNOWN].name

  const link = shallow(<AccountLink account={ACC_KNOWN} />)
  const nameEl = link.find('Link > span')

  expect(nameEl.text()).toEqual(anchorName)
  expect(nameEl.props().style.fontStyle).toEqual('italic')
  expect(link.getElements()).toMatchSnapshot()
})
