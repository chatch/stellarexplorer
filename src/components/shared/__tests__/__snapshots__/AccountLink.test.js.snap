// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders anchor name in italics if account a known anchor 1`] = `
Array [
  <span
    title="GATEMHCCKCY67ZUCKTROYN24ZYT5GK4EQZ65JJLDHKHRUZI3EUEKMTCH"
  >
    <Link
      replace={false}
      to="/account/GATEMHCCKCY67ZUCKTROYN24ZYT5GK4EQZ65JJLDHKHRUZI3EUEKMTCH"
    >
      <span
        style={
          Object {
            "fontStyle": "italic",
          }
        }
      >
        NaoBTC
      </span>
    </Link>
  </span>,
]
`;

exports[`renders short account for label when no label property 1`] = `
Array [
  <span
    title="GCGG3CIRBG2TTBR4HYZJ7JLDRFKZIYOAHFXRWLU62CA2QN52P2SUQNPJ"
  >
    <Link
      replace={false}
      to="/account/GCGG3CIRBG2TTBR4HYZJ7JLDRFKZIYOAHFXRWLU62CA2QN52P2SUQNPJ"
    >
      GCGG
    </Link>
  </span>,
]
`;

exports[`renders with label 1`] = `
Array [
  <span
    title="GCGG3CIRBG2TTBR4HYZJ7JLDRFKZIYOAHFXRWLU62CA2QN52P2SUQNPJ"
  >
    <Link
      replace={false}
      to="/account/GCGG3CIRBG2TTBR4HYZJ7JLDRFKZIYOAHFXRWLU62CA2QN52P2SUQNPJ"
    >
      Anchor
    </Link>
  </span>,
]
`;
